astropy/nddata/tests/test_ccddata.py

```diff
@@ -1196,7 +1197,12 @@ def test_write_read_with_flag_collection(tmp_path):
     assert len(ccd_after.flags.keys()) == len(ccd_data.flags.keys())
     assert ccd_after.flags.shape == ccd_data.flags.shape
     assert ccd_after.flags["BAD_PIXEL"].dtype == np.uint8
-    assert ccd_after.flags["SATURATED"].dtype == ">i8"
+    if sys.maxsize > 2**32:
```

Comment:
So, I don't understand why this check is different from the check right before it that does not seem to be a problem across archs.

```python
assert ccd_after.flags["BAD_PIXEL"].dtype == np.uint8
```

@BrunoSanchez are you able to clarify? Thanks.